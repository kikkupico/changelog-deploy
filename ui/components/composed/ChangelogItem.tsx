import dynamic from "next/dynamic";
import * as React from "react";
import { Circle, Props as CircleProps } from "../design/atoms/Circle";
import { deduplicate } from "../../utils/helpers";

const ReactMarkdown = dynamic<{ children: React.ReactNode }>(
  () => import("react-markdown") as any,
  { ssr: false }
);

type ChangeType = "bug-fix" | "feature" | "enhancement" | "misc";
const knownChangeTypes: ChangeType[] = ["bug-fix", "feature", "enhancement"];

type Props = {
  kind: string;
  types: string[];
  metadata:
    | {
        kind: "versioned";
        version: string;
        markdown: string;
      }
    | {
        kind: "update";
        version: string;
        shortDescription: string;
        longDescription: string;
      };
  date: Date;
  isLast: boolean;
  product: string;
  index: number;
};

const getReleaseKindText = (kind: Props["kind"]) => {
  switch (kind) {
    case "stable-release":
      return "Release";
    case "pre-release":
      return "Coming Soon";
    case "patch-release":
      return "Patch";
    default:
      return "Update";
  }
};

const getCircleColor = (index: number): CircleProps["color"] => {
  const colors: CircleProps["color"][] = [
    "brandCyan-400",
    "amber-500",
    "green-500",
    "indigo-500",
  ];
  return colors[index % 4] || colors[0];
};

export const getLabelText = (kind: Props["kind"], types: Props["types"]) => {
  return [getReleaseKindText(kind), ...deduplicate(types)].join(" â€¢ ");
};

export const getVersionPath = (offering: Props["product"], version: string) => {
  switch (offering) {
    case "cloud":
      return "/cloud/" + version;
    case "community-edition":
      return "/community-edition/" + version;
    case "enterprise-edition":
      return "/community-edition/" + version;
    default:
      return "/";
  }
};

export const ChangelogItem: React.FC<Props> = ({
  kind,
  types,
  metadata,
  date,
  isLast,
  index,
  product,
}) => {
  const changeTypes: ChangeType[] = types.map((t) => {
    if (knownChangeTypes.includes(t as ChangeType)) {
      return t as ChangeType;
    }
    return "misc";
  });

  const renderMarkdown = (text: string) => {
    return (
      <div className="prose prose-headings:text-gray-800 prose-headings:text-lg prose-headings:font-normal prose-headings:font-medium prose-p:text-gray-600 prose-p:font-light">
        <ReactMarkdown>{text}</ReactMarkdown>
        {text}
      </div>
    );
  };

  return (
    <div className="flex w-full">
      <div className="z-10">
        <Circle color={getCircleColor(index)} />
      </div>
      <div
        className={`${
          !isLast ? "border-l-2 border-gray-300" : ""
        } w-full pl-8 pt-1 pb-8 -ml-3`}
      >
        <div className="flex items-center mb-1">
          <p className="font-semibold text-xs">
            {getLabelText(kind, changeTypes)}
          </p>
          <p className="text-xs ml-auto">
            {date.toLocaleString("en-CA", {
              month: "short",
              day: "numeric",
              year: "numeric",
            })}
          </p>
        </div>
        {metadata.kind === "versioned" ? (
          <>
            <h2 className="text-lg text-gray-700 mb-2 font-semibold">
              <a
                className="hover:text-brandBlue-600"
                href={getVersionPath(product, metadata.version)}
              >
                {metadata.version}
              </a>
            </h2>
            {renderMarkdown(metadata.markdown)}
          </>
        ) : (
          <>
            <h2 className="text-lg text-gray-700 mb-2 font-semibold">
              <a
                className="hover:text-brandBlue-600"
                href={getVersionPath(product, metadata.version)}
              >
                {metadata.shortDescription}
              </a>
            </h2>
            {renderMarkdown(metadata.longDescription)}
          </>
        )}
      </div>
    </div>
  );
};
