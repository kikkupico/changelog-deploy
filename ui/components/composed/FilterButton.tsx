import * as React from "react";
import { Icon, Button } from "../design";

type Props = {
  onToggle: VoidFunction;
  filterActive: boolean;
};

export const FilterButton: React.FC<Props> = ({ onToggle, filterActive }) => {
  return (
    <Button
      size="sm"
      kind={filterActive ? "default-active" : "default"}
      id="def-with-icon"
      onClick={onToggle}
    >
      <Icon id="filter-icon-in-button" kind="filter" />
      &nbsp;{filterActive ? "Fewer Filters" : "More Filters"}
    </Button>
  );
};
