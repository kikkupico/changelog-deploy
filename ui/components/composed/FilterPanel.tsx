import * as React from "react";
import { useRouter, Router } from "next/router";
import { Label, DateInput } from "../design";
import {
  Filter,
  addFiltersToURLParams,
  removeFilterFromURLParam,
} from "../../utils/filters";

type Props = { filters: Record<Filter, string[] | undefined> };

const releaseTypes = [
  {
    type: "stable-release",
    label: "Release",
  },
  {
    type: "patch-release",
    label: "Patch",
  },
  {
    type: "update",
    label: "Update",
  },
  // {
  //   type: "pre-release",
  //   label: "Coming Soon",
  // },
];

const componentTypes = [
  {
    type: "server",
    label: "Server",
  },
  {
    type: "console",
    label: "Console",
  },
  {
    type: "cli",
    label: "CLI",
  },
];

// const tierTypes = [
//   {
//     type: "cloud-free",
//     label: "Free",
//   },
//   {
//     type: "cloud-standard",
//     label: "Standard",
//   },
//   {
//     type: "cloud-enterprise",
//     label: "Enterprise",
//   },
// ];

export const FilterPanel: React.FC<Props> = ({ filters }) => {
  const router = useRouter();

  return (
    <div>
      <div className="mb-2">
        <label className="block text-xs text-gray-400 font-semibold">
          Advanced Filters
        </label>
      </div>

      <div className="mb-2">
        <label className="block mb-1 text-xs">Core Release Type</label>
        <div className="flex items-center font-semibold text-sm space-x-2">
          {releaseTypes.map(({ type, label }) => {
            const isActive = filters["release-type"]?.includes(type);
            return (
              <Label
                id={`release-type-${type}`}
                key={type}
                onClick={() => {
                  let query: Router["query"] = {};
                  if (isActive) {
                    query = removeFilterFromURLParam(
                      router.query,
                      "release-type",
                      type
                    );
                  } else {
                    query = addFiltersToURLParams(
                      router.query,
                      "release-type",
                      type
                    );
                  }
                  router.push({
                    pathname: router.pathname,
                    query,
                  });
                }}
                kind={isActive ? "active" : "default"}
                text={label}
              />
            );
          })}
        </div>
      </div>

      <div className="mb-2">
        <label className="block mb-1 text-xs">Components</label>
        <div className="flex items-center font-semibold text-sm space-x-2">
          {componentTypes.map(({ type, label }) => {
            const isActive = filters["component"]?.includes(type);
            return (
              <Label
                id={`component-${type}`}
                key={type}
                onClick={() => {
                  let query: Router["query"] = {};
                  if (isActive) {
                    query = removeFilterFromURLParam(
                      router.query,
                      "component",
                      type
                    );
                  } else {
                    query = addFiltersToURLParams(
                      router.query,
                      "component",
                      type
                    );
                  }
                  router.push({
                    pathname: router.pathname,
                    query,
                  });
                }}
                kind={isActive ? "active" : "default"}
                text={label}
              />
            );
          })}
        </div>
      </div>

      {/*      <div className="mb-2">
        <label className="block mb-1 text-xs">Cloud Tiers</label>
        <div className="cursor-pointer flex items-center font-semibold text-sm space-x-2">
          {tierTypes.map(({ type, label }) => {
            const isActive = filters["tier"]?.includes(type);
            return (
              <Label
                id={`tier-${type}`}
                key={type}
                onClick={() => {
                  let query: Router["query"] = {};
                  if (isActive) {
                    query = removeFilterFromURLParam(
                      router.query,
                      "tier",
                      type
                    );
                  } else {
                    query = addFiltersToURLParams(router.query, "tier", type);
                  }
                  router.push({
                    pathname: router.pathname,
                    query,
                  });
                }}
                kind={isActive ? "active" : "default"}
                text={label}
              />
            );
          })}
        </div>
      </div>
*/}
      <div>
        <label className="block mb-1 text-xs">Date Filter</label>
        <div className="flex items-center space-x-2">
          <DateInput
            id={"date-from"}
            date={parseInt(filters.from?.join() || "")}
            onChange={(d) => {
              if (!d) {
                router.push({
                  pathname: router.pathname,
                  query: removeFilterFromURLParam(router.query, "from"),
                });
              } else {
                router.push({
                  pathname: router.pathname,
                  query: addFiltersToURLParams(
                    router.query,
                    "from",
                    d.getTime().toString(),
                    true
                  ),
                });
              }
            }}
          />
          <div className="w-6 h-0.5 bg-gray-400 rounded"></div>
          <DateInput
            id={"date-to"}
            date={parseInt(filters.to?.join() || "")}
            onChange={(d) => {
              if (!d) {
                router.push({
                  pathname: router.pathname,
                  query: removeFilterFromURLParam(router.query, "to"),
                });
              } else {
                router.push({
                  pathname: router.pathname,
                  query: addFiltersToURLParams(
                    router.query,
                    "to",
                    d.getTime().toString(),
                    true
                  ),
                });
              }
            }}
          />
        </div>
      </div>
    </div>
  );
};
