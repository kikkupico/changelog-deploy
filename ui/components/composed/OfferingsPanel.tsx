import * as React from "react";
import { Label } from "../design";
import { Offering } from "../../utils/useOfferings";

type Props = {
  selectedOffering: Offering;
  onSelectOffering: (o: Offering) => void;
};

const allOfferings: Offering[] = [
  "cloud",
  "community-edition",
  "enterprise-edition",
];
const getOfferingLabelText = (o: Offering) => {
  switch (o) {
    case "cloud":
      return "Hasura Cloud";
    case "community-edition":
      return "Hasura CE";
    case "enterprise-edition":
      return "Hasura EE";
  }
};

export const OfferingPanel: React.FC<Props> = ({
  selectedOffering,
  onSelectOffering,
}) => {
  return (
    <div className="flex items-center w-full">
      {allOfferings.map((o) => {
        return (
          <div className="mr-2" key={o}>
            <Label
              id={`offering-label-${o}`}
              kind={selectedOffering === o ? "active" : "default"}
              onClick={() => onSelectOffering(o)}
              text={getOfferingLabelText(o)}
            />
          </div>
        );
      })}
    </div>
  );
};
