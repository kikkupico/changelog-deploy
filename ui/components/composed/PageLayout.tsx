import * as React from "react";
import { Page, Header, Body, Footer, Button } from "../design";
import Image from "next/image";
import HasuraLogo from "../../public/images/hasura.svg";
import HasuraArtwork from "../../public/images/hasura-artwork.png";

export const PageLayout: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return (
    <Page>
      <Header>
        <div
          className={`
                        flex items-center
                        flex-col md:flex-row
                        justify-start md:justify-between 
                        w-full
                    `}
        >
          <div className="flex items-center mb-1 lg:mb-0">
            <Image
              src={HasuraLogo}
              className="w-28"
              width={"112px"}
              height={"33px"}
              alt="Hasura Logo"
            />
          </div>
          <Button kind="secondary" size="md" id="hasura-cloud-link">
            Create Your API with Hasura Cloud
          </Button>
        </div>
      </Header>
      <Body>{children}</Body>
      <Footer>
        <div className="w-full flex flex-col">
          <Image
            src={HasuraArtwork}
            className="md:w-8/12 mx-auto mb-4"
            alt="Hasura Artwork"
          />
          <div
            className={`
                        flex items-center
                        flex-col md:flex-row
                        justify-start md:justify-between 
                        w-full
                    `}
          >
            <div className="flex items-center mb-1 lg:mb-0">
              <Image src={HasuraLogo} alt="Hasura Logo" />
            </div>
            <p className="text-sm text-gray-600">
              Â© {new Date().getFullYear()} Hasura Inc. All rights reserved
            </p>
          </div>
        </div>
      </Footer>
    </Page>
  );
};
