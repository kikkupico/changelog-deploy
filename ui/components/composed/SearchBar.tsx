import * as React from "react";
import { Button, Icon } from "../design";

type Props = {
  id: string;
  search?: string;
  placeholder?: string;
  onSearchChange?: (s: string) => void;
  onSearch?: (s: string) => void;
};

export const SearchBar: React.FC<Props> = ({
  id,
  search = "",
  onSearchChange,
  onSearch,
  placeholder = "",
}) => {
  const [hasFocus, setHasFocus] = React.useState(false);
  return (
    <div
      id={id}
      data-testid={id}
      className={`
            w-full
            flex flex-row items-center h-12 rounded
            px-2
            ${
              hasFocus
                ? "border-2 border-brandBlue-400 ring-brandBlue-400"
                : "border border-gray-400 hover:border-gray-500"
            }
            hover:shadow 
            
        `}
    >
      <div className="flex w-5/6">
        <div className="mr-2 flex items-center">
          <Icon id={`${id}-search-icon`} kind="search" />
        </div>
        <input
          id={`${id}-text-input`}
          data-testid={`${id}-text-input`}
          placeholder={placeholder}
          value={search}
          className="border-none outline-none w-full"
          onFocus={() => {
            setHasFocus(true);
          }}
          onBlur={() => {
            setHasFocus(false);
          }}
          onChange={(e) => {
            if (onSearchChange) {
              onSearchChange(e.target.value);
            }
          }}
        />
      </div>
      <div className="flex flex-row items-center justify-end w-1/6">
        <Button
          id={`${id}-search-button`}
          kind="primary"
          size="sm"
          onClick={() => {
            if (onSearch) {
              onSearch(search || "");
            }
          }}
        >
          Search
        </Button>
      </div>
    </div>
  );
};
