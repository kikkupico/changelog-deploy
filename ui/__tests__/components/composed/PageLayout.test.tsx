import { render } from "@testing-library/react";
import { PageLayout } from "../../../components/composed/PageLayout";
import "@testing-library/jest-dom";

// used snapshoht here because layout barely ever changes

test("Matches PageLayout Snapshot", () => {
  const pageLayout = render(
    <PageLayout>
      <div>Test children</div>
    </PageLayout>
  );
  expect(pageLayout).toMatchSnapshot();
});
