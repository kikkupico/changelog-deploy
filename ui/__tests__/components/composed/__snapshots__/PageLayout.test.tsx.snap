// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches PageLayout Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative flex flex-col w-full min-h-screen items-center"
      >
        <header
          class="flex w-full items-center max-w-screen-lg p-4"
        >
          <div
            class="
                        flex items-center
                        flex-col md:flex-row
                        justify-start md:justify-between 
                        w-full
                    "
          >
            <div
              class="flex items-center mb-1 lg:mb-0"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27112%27%20height=%2733%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Hasura Logo"
                  class="w-28"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <button
              class="flex items-center w-fit focus:outline-none rounded text-brandBlue-400 border border-brandBlue-400 hover:bg-brandBlue-100 h-9 px-2 text-md"
              data-testid="hasura-cloud-link"
              id="hasura-cloud-link"
            >
              Create Your API with Hasura Cloud
            </button>
          </div>
        </header>
        <main
          class="flex flex-col w-full max-w-screen-lg p-4"
        >
          <div>
            Test children
          </div>
        </main>
        <footer
          class="flex w-full h-auto max-w-screen-lg p-4"
        >
          <div
            class="w-full flex flex-col"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Hasura Artwork"
                class="md:w-8/12 mx-auto mb-4"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div
              class="
                        flex items-center
                        flex-col md:flex-row
                        justify-start md:justify-between 
                        w-full
                    "
            >
              <div
                class="flex items-center mb-1 lg:mb-0"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Hasura Logo"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="text-sm text-gray-600"
              >
                Â© 
                2022
                 Hasura Inc. All rights reserved
              </p>
            </div>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative flex flex-col w-full min-h-screen items-center"
    >
      <header
        class="flex w-full items-center max-w-screen-lg p-4"
      >
        <div
          class="
                        flex items-center
                        flex-col md:flex-row
                        justify-start md:justify-between 
                        w-full
                    "
        >
          <div
            class="flex items-center mb-1 lg:mb-0"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27112%27%20height=%2733%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Hasura Logo"
                class="w-28"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <button
            class="flex items-center w-fit focus:outline-none rounded text-brandBlue-400 border border-brandBlue-400 hover:bg-brandBlue-100 h-9 px-2 text-md"
            data-testid="hasura-cloud-link"
            id="hasura-cloud-link"
          >
            Create Your API with Hasura Cloud
          </button>
        </div>
      </header>
      <main
        class="flex flex-col w-full max-w-screen-lg p-4"
      >
        <div>
          Test children
        </div>
      </main>
      <footer
        class="flex w-full h-auto max-w-screen-lg p-4"
      >
        <div
          class="w-full flex flex-col"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Hasura Artwork"
              class="md:w-8/12 mx-auto mb-4"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="
                        flex items-center
                        flex-col md:flex-row
                        justify-start md:justify-between 
                        w-full
                    "
          >
            <div
              class="flex items-center mb-1 lg:mb-0"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Hasura Logo"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <p
              class="text-sm text-gray-600"
            >
              Â© 
              2022
               Hasura Inc. All rights reserved
            </p>
          </div>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
